lambda/audit.py

```diff
@@ -0,0 +1,319 @@
+import json
+import os
+import uuid
+
+import boto3
+from botocore.exceptions import ClientError
+
+import github_api
+from logger import LOG
+
+
+def start(message=None):
+    """ Start an audit of github membership """
+    org = os.environ.get("GITHUB_ORG")
+    audit_id = str(uuid.uuid4())
+    LOG.info({"action": "Starting github audit", "org": org, "audit_id": audit_id})
+
+    sns_org_members = create_sns_message(
+        {"action": "log_org_membership", "org": org, "audit_id": audit_id}
+    )
+    publish_alert(sns_org_members)
+
+    sns_org_repos = create_sns_message(
+        {"action": "log_org_repos", "org": org, "audit_id": audit_id}
+    )
+    publish_alert(sns_org_repos)
+
+    sns_org_teams = create_sns_message(
+        {"action": "log_org_teams", "org": org, "audit_id": audit_id}
+    )
+    publish_alert(sns_org_teams)
+    return True
+
+
+def log_org_membership(message=None):
+    """ Audit github organization membership """
+    org = os.environ.get("GITHUB_ORG")
```

Comment:
Should this parameter come from the message instead of an environment variable? Is looks like the org is included in the SNS message already.

Reply:
Probably yes. At the moment it's setup to audit alphagov from an env var and gets an alphagov PAT from SSM. I guess if we had a case for auditing another org we could refactor those things to be configurable - in which case making this change through the subsequent SNS invokes now would make that easier later.

Reply:
Maybe this should be turned into an issue for later.

Reply:
Either way, I think that most of these `os.environ.get(name)` lines should probably be `os.environ[name]`, as the function can't work without this, so we might as well blow up early, with a good `NotFound` error?

Reply:
Lets, change this in the future if we need to expand it to more github orgs.